### PR TITLE
feat(route): add UK Parliament Petitions

### LINKED ISSUE #17708: 【petition 英国】英国请愿官网
### 类型

政务消息

### 网站地址

https://petition.parliament.uk/petitions

### 网站描述

英国请愿官网 At 100,000 signatures your petition on the UK Government and Parliament site will be considered for a debate in Parliament.

https://petition.parliament.uk/help

### 需要生成什么内容？

请愿按签名数倒序和按时间倒序

https://petition.parliament.uk/petitions?state=open

https://petition.parliament.uk/petitions?state=recent

json格式

https://petition.parliament.uk/petitions.json?page=1&state=recent

https://petition.parliament.uk/petitions.json?page=1&state=open

### 额外描述

# 感谢

### 这不是重复的 RSS 请求

- [X] 我已经搜索了[现有 issue](https://github.com/DIYgod/RSSHub/issues) 和 [pull requests](https://github.com/DIYgod/RSSHub/pulls)，以确保该 RSS 尚未被请求。

### PR DESCRIPTION
**Pull request by nczitzk**
## Involved Issue / 该 PR 相关 Issue

Close #17708

## Example for the Proposed Route(s) / 路由地址示例

```routes
/parliament.uk/petitions/all
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [x] Parsed / 可以解析
  - [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/routes/parliament.uk/petitions.ts [hunk 1]
```diff
@@ -0,0 +1,190 @@
+import path from 'node:path';
+import { type Context } from 'hono';
+import { ofetch } from 'ofetch';
```
Comment (TonyRL) [inline lib/routes/parliament.uk/petitions.ts:3]:
```suggestion
import ofetch from '@/utils/ofetch';
```

File: lib/routes/parliament.uk/petitions.ts [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    const rootUrl: string = 'https://petition.parliament.uk';
+    const targetUrl: string = new URL(`petitions?state=${state}`, rootUrl).href;
+    const jsonUrl: string = new URL('petitions.json', rootUrl).href;
+
+    const response = await ofetch(targetUrl);
+    const $: CheerioAPI = load(response);
+    const language: string = $('html').prop('lang') ?? 'en';
+
+    const jsonResponse = await ofetch(jsonUrl, {
+        searchParams: {
```
Comment (TonyRL) [inline lib/routes/parliament.uk/petitions.ts:26]:
```suggestion
        query: {
```

File: lib/routes/parliament.uk/petitions.ts [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+        allowEmpty: true,
+        image: feedImage,
+        author: $('meta[name="msapplication-tooltip"]').prop('content'),
+        language,
+        id: $('meta[property="og:url"]').prop('content'),
+    };
+};
+
+export const route: Route = {
+    path: '/petitions/:state{.+}?',
```
Comment (TonyRL) [inline lib/routes/parliament.uk/petitions.ts:87]:
```suggestion
    path: '/petitions/:state?',
```

File: lib/routes/parliament.uk/petitions.ts (hunk 1 above)
Comment (nczitzk) [reply lib/routes/parliament.uk/petitions.ts:3]: Thanks for the review. ☺️

**Pull request merged**